### ISSUE TITLE
Errors and warnings not being reported

### ISSUE THREAD

**Issue by willpracht** [Microsoft/vscode-tslint#254]
After a fresh install of this extension, neither errors nor warnings are being reported.

I get the following in the "Output" pane:

```
[Info  - 09:47:51] Linter is running.
[Info  - 09:47:52] TSLint library loaded from: /Users/me/projects/my-project/node_modules/tslint/lib/tslint.js
```

So, it seems that it's being loaded correctly. I have a tslint.json in my project root. If I run tslint via a gulp task I've setup, linting works just fine.

I'm running TypeScript 1.8.10 globally: `/Users/me/.nvm/versions/node/v4.8.4/bin/tsc`.

Am I missing something?

**Comment by edarev** [comment]
I have the same problem. Any solutions?

**Comment by egamma** [comment]
@willpracht 
what versions of tslint and vscode-tslint are you running? 
Is tslint also installed globally, same as typescript?

What output do you see when you enable tracing? To enable tracing add the setting "tslint.trace.server": "verbose"

**Comment by DjulLau** [comment]
I have a similar problem except I get errors but no warnings...
vscode 1.16.0
vscode-tslint 1.0.3
typescript 2.5.2 (from vscode)
I get the warnings in the console during the build process but not when live editing in vscode.

Tracing output example :
```
[Trace - 12:03:30] Sending notification 'textDocument/didChange'.
Params: {
    "textDocument": {
        "uri": "file:///z%3A/CBi/cbi-modsho/dev-ionic/src/core/shared/player/player.component.ts",
        "version": 2
    },
    "contentChanges": [
        {
            "text": "import { Component, ViewChild, OnInit, Input, Output, EventEmitter } from \"@angular/core\";\r\nimport { NavParams, Slides, AlertController } from \"ionic-angular\";\r\n\r\nimport { PackageService, PackageType } from \"../application-logic/package.service\";\r\nimport { SharingService } from \"../application-logic/sharing.service\";\r\nimport { ConfigurationService } from \"../application-logic/configuration.service\";\r\n// import { AnalyticsService } from \"../../shared/analytics/analytics.service\";\r\nimport { Trackers } from \"../../shared/analytics/trackers\";\r\n\r\nimport { PlayerSlidePositionState } from \"./player-slide.component\";\r\n\r\nimport { PresentationModel } from \"../models/presentation.model\";\r\nimport { ModuleModel } from \"../models/module.model\";\r\nimport { PageModel } from \"../models/page.model\";\r\nimport { GalleryPackageModel } from \"../models/gallery-package.model\";\r\n\r\nimport { IModshoSlideListener, ModshoPlayerType } from \"../modsho-slide-wrapper\";\r\n\r\ndeclare var global: { require: Function, process: { cwd: Function } };\r\n\r\ninterface ISlideWindow extends Window {\r\n    slide;\r\n}\r\n\r\n@Component({\r\n    selector: \"player-component\",\r\n    templateUrl: \"player.component.html\",\r\n})\r\n\r\nexport class PlayerComponent implements OnInit, IModshoSlideListener {\r\n    @Input(\"playerType\") public playerType: ModshoPlayerType;\r\n    @Input(\"presentation\") public presentation: PresentationModel;\r\n    @Output(\"openPresentation\") public openPresentation = new EventEmitter<PresentationModel>();\r\n    @ViewChild(\"slider\") public slider: Slides;\r\n\r\n    public pages: PageModel[] = [];\r\n    public customDataForPages: string[] = [];\r\n\r\n    private _pannelOpened: boolean = false;\r\n    private _currentPageIndex: number;\r\n    private currentModule: ModuleModel;\r\n    private open;\r\n\r\n    constructor(\r\n        private sharingService: SharingService,\r\n\r\n        private pkgService: PackageService,\r\n        private alertCtrl: AlertController,\r\n        private analyticsService: AnalyticsService,\r\n        configService: ConfigurationService) {\r\n        this._pannelOpened = false;\r\n        if (configService.nwjs) {\r\n            this.open = global[\"require\"](\"open\");\r\n            let test;\r\n        }\r\n    }\r\n\r\n    public ngOnInit() {\r\n        // Init\r\n        this.presentation.modules.forEach(module => {\r\n            module.pages.forEach(page => {\r\n                this.pages.push(page);\r\n                this.customDataForPages.push(this.presentation.customDataForModuleAndPage(module.id, page.id));\r\n            });\r\n        });\r\n        if (this.slider) {\r\n            this.slider.pager = false;\r\n            this.slider.centeredSlides = true;\r\n            this.slider.slideToClickedSlide = true;\r\n            // this.slider.effect = \"coverflow\";\r\n            // this.slider.coverflow = this.coverflowSettings;\r\n        }\r\n\r\n        this.currentPageIndex = 0;\r\n\r\n        if (this.presentation.modules.length === 0) {\r\n            this.alertCtrl.create({\r\n                title: \"Présentation vide !\",\r\n                message: \"Rendez-vous dans l'éditeur de présentation pour y ajouter des modules.\",\r\n                enableBackdropDismiss: true,\r\n                buttons: [{\r\n                    text: \"OK\",\r\n                    role: \"OK\",\r\n                }],\r\n            }).present();\r\n        }\r\n    }\r\n\r\n    public get pannelOpened() {\r\n        return this._pannelOpened;\r\n    }\r\n\r\n    public set pannelOpened(value: boolean) {\r\n        this._pannelOpened = value;\r\n    }\r\n\r\n    public PannelOnDoubleTap(e) {\r\n        this._pannelOpened = !this._pannelOpened;\r\n    }\r\n\r\n    public slideDidChange() {\r\n        this.currentPageIndex = this.slider.realIndex;\r\n    }\r\n\r\n    public slideState(slideIndex: number): PlayerSlidePositionState {\r\n        let realIndex = this.slider.realIndex;\r\n        if (slideIndex === realIndex) {\r\n            return PlayerSlidePositionState.Current;\r\n        }\r\n        if ((slideIndex === realIndex - 1) || (slideIndex === realIndex + 1)) {\r\n            return PlayerSlidePositionState.Hot;\r\n        }\r\n        return PlayerSlidePositionState.Cold;\r\n    }\r\n\r\n    public pannelButtonClicked() {\r\n        this._pannelOpened = !this._pannelOpened;\r\n    }\r\n\r\n    public slideClicked(pageIndex: number) {\r\n        this.currentPageIndex = pageIndex;\r\n        this._pannelOpened = false;\r\n        this.slider.slideTo(pageIndex, 0, false);\r\n    }\r\n\r\n    public contentClick(event) {\r\n        if (this._pannelOpened) {\r\n            this._pannelOpened = false;\r\n        }\r\n    }\r\n\r\n    public customDataForPageIndex(pageIndex: number) {\r\n        return this.customDataForPages[pageIndex];\r\n    }\r\n\r\n    // IModshoSlideListener\r\n    public modshoSlideListenerOpenFile(path: string) {\r\n        if (this.open) {\r\n            if (path.indexOf(this.pkgService.getInstalledPackageFolderPath(PackageType.CommonGallery)) === -1 &&\r\n                path.indexOf(this.pkgService.getInstalledPackageFolderPath(PackageType.PersonalGallery)) === -1) {\r\n                // Build path relative to html page\r\n                let page = this.pages[this._currentPageIndex];\r\n                let pagePath = this.pkgService.getInstalledPackageFilePath(PackageType.Presentations, page.index);\r\n                let pagePathComp = pagePath.split(\"/\");\r\n                pagePathComp.pop();\r\n                let pathComp = path.split(\"/\");\r\n                while (pathComp[0] === \"..\") {\r\n                    pathComp.shift();\r\n                    pagePathComp.pop();\r\n                }\r\n                path = pagePathComp.join(\"/\").concat(pathComp.join(\"/\"));\r\n            }\r\n            this.open(path);\r\n        }\r\n    }\r\n\r\n    public modshoSlideListenerOpenMedia(pkgType: PackageType, id: number, useZoom: boolean) {\r\n        let pkg = <GalleryPackageModel>this.pkgService.loadedPackage(pkgType);\r\n        if (!pkg) {\r\n            console.warn(\"modshoSlideListenerOpenMedia:: package not installed\", pkgType);\r\n            return;\r\n        }\r\n\r\n        let media = pkg.getMediaById(id);\r\n        if (!media) {\r\n            console.warn(\"modshoSlideListenerOpenMedia:: media not found in package\", pkgType, id);\r\n            return;\r\n        }\r\n\r\n        let path = this.pkgService.getInstalledPackageFilePath(pkgType, useZoom ? media.zoom : media.file);\r\n        this.open(path);\r\n    }\r\n\r\n    public modshoSlideListenerJumpToPage(pageIndex: number) {\r\n        if (pageIndex < 0 || pageIndex >= this.currentModule.pages.length) {\r\n            console.error(\"Invalid argument : page index must be between 0 and pages count - 1\");\r\n            return;\r\n        }\r\n\r\n        let linearPageIndex = 0;\r\n        this.presentation.modules.some(module => {\r\n            if (module !== this.currentModule) {\r\n                linearPageIndex += module.pages.length;\r\n                return false;\r\n            }\r\n            return true;\r\n        });\r\n\r\n        this.slideClicked(linearPageIndex);\r\n    }\r\n\r\n    public modshoSlideListenerJumpToModulePage(moduleId: number, pageIndex: number) {\r\n        // Search module to open in package\r\n        let module = this.pkgService.presentationPackage.modules.find(m => { return m.id === moduleId; });\r\n        if (!module) {\r\n            console.error(`Invalid argument : module with id ${moduleId} not found in package`);\r\n            return;\r\n        }\r\n\r\n        // Make sure page is within bounds\r\n        if (pageIndex < 0 || pageIndex >= module.pages.length) {\r\n            console.error(\"Invalid argument : page index must be between 0 and pages count - 1\");\r\n            return;\r\n        }\r\n\r\n        // Search module to open in current presentation\r\n        if (this.presentation.modules.indexOf(module) !== -1) {\r\n            // Module found in current presentation\r\n            let linearPageIndex = 0;\r\n            this.presentation.modules.some(m => {\r\n                if (m !== module) {\r\n                    linearPageIndex += m.pages.length;\r\n                    return false;\r\n                }\r\n                return true;\r\n            });\r\n            linearPageIndex += pageIndex;\r\n\r\n            this.slideClicked(linearPageIndex);\r\n        } else {\r\n            // Module not found in current presentation\r\n            // Create a temp presentation and open in popup\r\n            let presentation = new PresentationModel();\r\n            presentation.name = module.name;\r\n            presentation.id = -1;\r\n            presentation.modules.push(module);\r\n\r\n            this.openPresentation.emit(presentation);\r\n            // this.appService.openPresentation(presentation);\r\n        }\r\n    }\r\n    //\r\n\r\n    public set currentPageIndex(index: number) {\r\n        this._currentPageIndex = index;\r\n\r\n        // Get current module\r\n        let pagesCount = 0;\r\n        this.presentation.modules.some(module => {\r\n            if (pagesCount + module.pages.length > this._currentPageIndex) {\r\n                this.currentModule = module;\r\n                return true;\r\n            } else {\r\n                pagesCount += module.pages.length;\r\n                return false;\r\n            }\r\n        });\r\n\r\n    }\r\n\r\n    public get currentPageIndex() {\r\n        return this._currentPageIndex;\r\n    }\r\n}\r\n"
        }
    ]
}


[Trace - 12:03:30] Sending notification 'textDocument/didSave'.
Params: {
    "textDocument": {
        "uri": "file:///z%3A/CBi/cbi-modsho/dev-ionic/src/core/shared/player/player.component.ts",
        "version": 2
    }
}


[Trace - 12:03:30] Received request 'workspace/configuration - (5)'.
Params: {
    "items": [
        {
            "scopeUri": "file:///z%3A/CBi/cbi-modsho/dev-ionic/src/core/shared/player/player.component.ts",
            "section": "tslint"
        }
    ]
}


[Trace - 12:03:30] Sending response 'workspace/configuration - (5)'. Processing request took 0ms
[Trace - 12:03:30] Received request 'workspace/configuration - (6)'.
Params: {
    "items": [
        {
            "scopeUri": "file:///z%3A/CBi/cbi-modsho/dev-ionic/src/core/shared/player/player.component.ts",
            "section": "tslint"
        }
    ]
}


[Trace - 12:03:30] Sending response 'workspace/configuration - (6)'. Processing request took 0ms
[Trace - 12:03:30] Received request 'workspace/configuration - (7)'.
Params: {
    "items": [
        {
            "scopeUri": "file:///z%3A/CBi/cbi-modsho/dev-ionic/src/core/shared/player/player.component.ts",
            "section": "tslint"
        }
    ]
}


[Trace - 12:03:30] Sending response 'workspace/configuration - (7)'. Processing request took 0ms
[Trace - 12:03:30] Received notification 'tslint/status'.
Params: {
    "state": 1
}


[Trace - 12:03:30] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///z%3A/CBi/cbi-modsho/dev-ionic/src/core/shared/player/player.component.ts",
    "diagnostics": []
}
```

**Comment by egamma** [comment]
@DjulLau the trace shows the file has been analyzed but no warnings were identified:
- what version tslint do you have installed? 
- Can you share your tslint.json? 
- Do have configured tslint specific settings in your user settings?

>I have a similar problem except I get errors
what errors do you get and where do you see them.

**Comment by willpracht** [comment]
@egamma - 

tslint: /Users/me/.nvm/versions/node/v4.8.4/bin/tslint  :   5.7.0
typescript: /Users/me/.nvm/versions/node/v4.8.4/bin/tsc    :    2.5.2

vscode: 1.17.0-insider
vscode-tslint: 1.0.3

Locally, I am running TypeScript 1.8.10, but I just assumed it would either run via my local or my global, either way it should still work, I thought. 

After enabling trace, I see the following for a file I have open with a known lint error:
```
[Trace - 10:47:07] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///Users/me/projects/my-project/app/components/whatever/whatever.states.ts",
    "diagnostics": []
}
```

**Comment by willpracht** [comment]
@egamma I noticed that if I have the exact same setup in vscode (non-insiders build) it seems to work perfectly.

Is it something with vscode-insiders that disallows this from working?

**Comment by DjulLau** [comment]
@egamma 

- global tslint is 5.7.0
- tslint.json
```json
{
  "extends": "tslint:latest",
  "rules": {
    "quotemark": [true, "double"],
    "no-bitwise": false,
    "no-constructor-vars": false,
    "max-line-length": [false, 120],
    "object-literal-sort-keys": false,
    "no-string-literal": false,
    "no-console": [false, "log", "error"],
    "no-any": false,
    "ordered-imports": [false],
    "eofline": false,
    "member-access": true,
    "whitespace": [true, "check-branch","check-decl","check-operator","check-module","check-separator","check-type"/*,"check-typecast"*/],
    "variable-name": [true, "ban-keywords", "check-format", "allow-leading-underscore"]
  },
  "rulesDirectory": [
    "node_modules/tslint-eslint-rules/dist/rules"
  ]
}
```
- no specific user settings

> what errors do you get and where do you see them.

In VSCode I see errors as normal (bottom left icon + editor + problems panel) but no warning.

For example:
```typescript
export class Test {
    public test() {
        let s: string = 0;
        let t;
    }
}
```
Generates
```
[Trace - 18:34:12] Sending notification 'textDocument/didChange'.
Params: {
    "textDocument": {
        "uri": "file:///z%3A/CBi/cbi-modsho/dev-ionic/src/core/shared/analytics/test.ts",
        "version": 68
    },
    "contentChanges": [
        {
            "text": "export class Test {\r\n    public test() {\r\n        let s: string = 0;\r\n        let t;\r\n    }\r\n}"
        }
    ]
}


[Trace - 18:34:12] Received request 'workspace/configuration - (2435)'.
Params: {
    "items": [
        {
            "scopeUri": "file:///z%3A/CBi/cbi-modsho/dev-ionic/src/core/shared/analytics/test.ts",
            "section": "tslint"
        }
    ]
}


[Trace - 18:34:12] Sending response 'workspace/configuration - (2435)'. Processing request took 1ms
[Trace - 18:34:13] Received request 'workspace/configuration - (2436)'.
Params: {
    "items": [
        {
            "scopeUri": "file:///z%3A/CBi/cbi-modsho/dev-ionic/src/core/shared/analytics/test.ts",
            "section": "tslint"
        }
    ]
}


[Trace - 18:34:13] Sending response 'workspace/configuration - (2436)'. Processing request took 0ms
[Trace - 18:34:13] Received notification 'tslint/status'.
Params: {
    "state": 1
}


[Trace - 18:34:13] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///z%3A/CBi/cbi-modsho/dev-ionic/src/core/shared/analytics/test.ts",
    "diagnostics": []
}
```
And I see one error in VSCode : 
>file: 'file:///z%3A/CBi/cbi-modsho/dev-ionic/src/core/shared/analytics/test.ts'
severity: 'Erreur'
message: 'Type '0' is not assignable to type 'string'.'
at: '3,13'
source: 'ts'

But no "unused variable" warning as I used to see (same for all warnings).
I've also noticed that even if I deactivate the plugin I still got the error so I'm obviously missing something.

Everything was working with the same user settings and tslint.json till recently. 
In the meantime I remember having installed vscode-insiders too (by mistake) but uninstalled it and got the stable 64 bits version instead. Not sure if it is related.

**Comment by egamma** [comment]
@willpracht 

>Is it something with vscode-insiders that disallows this from working?
I use it in vscode-insiders and linting works fine. Can you please try whether my test setup works for you. It takes less than 5' to run:
- git clone https://github.com/Microsoft/vscode-tslint.git
- cd vscode-tslint/tslint-tests
- npm install
- code or code-insiders .
- open tests/array-type.ts in the editor
=> you should see a tslint warning

If this is working then we can look into what is different in your setup.

**Comment by egamma** [comment]
@DjulLau 

> But no "unused variable" warning as I used to see (same for all warnings).

Have you upgraded tslint recently? Notice that the no unused variable rule in tslint version 5 now requires type information. vscode-tslint does not support rules that require type information, pls see https://github.com/Microsoft/vscode-tslint/issues/70#issuecomment-241041929. Therefore you will not see this warning anymore. The recommendation is to use corresponding TS compiler option.

I see that in your tslint.json you have enabled the 'quotemark' rule, can you test whether a quotemark rule violation is reported as a warning?

Regarding the error you are seeing. 

>file: 'file:///z%3A/CBi/cbi-modsho/dev-ionic/src/core/shared/analytics/test.ts'
severity: 'Erreur'
message: 'Type '0' is not assignable to type 'string'.'
at: '3,13'
source: 'ts'

This error is reported by the TypeScript language service (source === 'ts') and it isn't a warning reported by tslint (source === 'tslint').

**Comment by DjulLau** [comment]
So I was confusing TypeScript errors and tslint error sorry. Thanks for clearing that.
Still quotemark rule and all other tslint rules I had working before didn't work anymore.

I went thru your test and it worked OK so I though it had to do with my project configuration.
I tried a fresh checkout / npm install of it with no luck.

tslint was not installed locally (not in project package.json but was installed globally) so I installed it locally and it worked! Then I removed it locally and... it's still working o_O

**Comment by egamma** [comment]
@DjulLau good that it is working, but I don't fully understand the difference between what did not work. There must be an issue in here. 

Do I have this right:
- tslint and typescript installed globally - did not work  
- tslint and typescript installed locally - worked  - Do you know whether the version of tslint changed?
- removed tslint locally - still works

When you say npm install did you also npm install typescript with tslint? Remember that tslint requires typescript as a peer dependency.

**Comment by DjulLau** [comment]
- tslint installed globally (5.7.0), typescript installed globally (2.5.2) and locally (2.2.2) - did not work
- added tslint locally (4.5.1) - did work
- updated tslint locally (5.7.0) - still worked
- removed tslint locally - still worked

As if the first local installation of tslint fixed something...

When I say 
>I tried a fresh checkout / npm install of it with no luck.

I mean npm install my project so the whole package.json (with typescript as a devDependencies but no tslint at this time).

**Comment by willpracht** [comment]
@egamma 

I'm going to throw a wrench into things a bit! ;p

I had both code and code-insiders installed. Couldn't get anything to work with insiders.

I switched back over to code, tslint seemed to be working just fine. I had no configuration options set for it.

So I thought, "cool, I don't need to worry about insiders anymore, let me just remove that, stick with code from now on."

I removed insiders, removed code, blew away stored configs and extensions for code, then reinstalled code.

Installed vscode-tslint, now it doesn't work again. 

So thoroughly confused. :)

**Comment by egamma** [comment]
@willpracht Could you verify that my test-setup mentioned above works in both "code" and "code-insiders"?

>I'm running TypeScript 1.8.10 globally

This could be a problem since tslint v5.7 requires Typescript > 2.1 as a peer dependency, 

I'll make a version of the extension available with more tracing information. I'll let you know when it is available.

**Comment by egamma** [comment]
@willpracht @edarev I've published a new version of the extension which can emit more trace information. Can you please send me a trace when tslint fails for you. To do so:

- Define the user setting `tslint.trace.server` : "verbose"
- Restart vscode
- Open the .ts file where linting fails
- Open the tslint output channel by clicking the TSLint item in the status line
- Attach the output to this issue

**Comment by edarev** [comment]
@egamma I add this `config/tslint.json` in C:\Program Files\Microsoft VS Code: 
```
{
  /*
   * Possible values:
   * - the name of a built-in config
   * - the name of an NPM module which has a "main" file that exports a config object
   * - a relative path to a JSON file
   */
  "extends": "tslint:recommended",
  "rules": {
    /*
     * Any rules specified here will override those from the base config we are extending
     */
    "no-constructor-vars": true,
    "no-any": true,
    "typedef": [true,
      "call-signature", "arrow-call-signature",
      "parameter", "arrow-parameter",
      "property-declaration", "variable-declaration",
      "member-variable-declaration"]
  },
  "rulesDirectory": [
    /*
     * A list of relative or absolute paths to directories that contain custom rules.
     * See the Custom Rules documentation below for more details.
     */
  ]
}

```
In VS code i add this settings (`.vscode/settings.json`):
```
{
    "tslint.enable": true,
    "tslint.configFile": "config/tslint.json",
    "tslint.trace.server": "verbose",
    "typescript.tsdk": "node_modules\\typescript\\lib"
}
```
In my project i have the same tslint.json like in C:\Program Files\Microsoft VS Code.

I add some blank-lines in the code: 

```
private showKeyboardIsOn: boolean = false;


   
private showCapsIsOn: boolean = false;
```
And i don't see any errors in warnings panel. But if i write `npm run tslint`, this command show me errors in `tslint.error.log`. 

This is my trace

```
[Trace - 10:04:45] tslint configuration:
{ rulesDirectory: [],
  rules: 
   { align: [ true, 'parameters', 'statements' ],
     'class-name': true,
     'comment-format': [ true, 'check-space' ],
     curly: true,
     eofline: true,
     forin: true,
     indent: [ true, 'spaces' ],
     'interface-name': [ true, 'always-prefix' ],
     'jsdoc-format': true,
     'label-position': true,
     'label-undefined': true,
     'max-line-length': [ true, 120 ],
     'member-access': true,
     'member-ordering': [ true, { order: 'statics-first' } ],
     'new-parens': true,
     'no-any': true,
     'no-arg': true,
     'no-bitwise': true,
     'no-conditional-assignment': true,
     'no-consecutive-blank-lines': true,
     'no-console': [ true, 'debug', 'info', 'log', 'time', 'timeEnd', 'trace' ],
     'no-construct': true,
     'no-constructor-vars': true,
     'no-debugger': true,
     'no-duplicate-key': true,
     'no-duplicate-variable': true,
     'no-empty': true,
     'no-eval': true,
     'no-internal-module': true,
     'no-namespace': true,
     'no-reference': true,
     'no-shadowed-variable': true,
     'no-string-literal': true,
     'no-switch-case-fall-through': false,
     'no-trailing-whitespace': true,
     'no-unreachable': true,
     'no-unused-expression': true,
     'no-unused-new': true,
     'no-unused-variable': [ true, 'react' ],
     'no-use-before-declare': false,
     'no-var-keyword': true,
     'no-var-requires': true,
     'object-literal-sort-keys': true,
     'one-line': 
      [ true,
        'check-catch',
        'check-else',
        'check-finally',
        'check-open-brace',
        'check-whitespace' ],
     'one-variable-per-declaration': [ true, 'ignore-for-loop' ],
     quotemark: [ true, 'double', 'avoid-escape' ],
     radix: true,
     semicolon: [ true, 'always' ],
     'switch-default': true,
     'trailing-comma': [ true, { singleline: 'never', multiline: 'always' } ],
     'triple-equals': [ true, 'allow-null-check' ],
     typedef: 
      [ true,
        'call-signature',
        'arrow-call-signature',
        'parameter',
        'arrow-parameter',
        'property-declaration',
        'variable-declaration',
        'member-variable-declaration' ],
     'typedef-whitespace': 
      [ true,
        { 'call-signature': 'nospace',
          'index-signature': 'nospace',
          parameter: 'nospace',
          'property-declaration': 'nospace',
          'variable-declaration': 'nospace' },
        { 'call-signature': 'onespace',
          'index-signature': 'onespace',
          parameter: 'onespace',
          'property-declaration': 'onespace',
          'variable-declaration': 'onespace' } ],
     'use-isnan': true,
     'variable-name': [ true, 'ban-keywords', 'check-format', 'allow-pascal-case' ],
     whitespace: 
      [ true,
        'check-branch',
        'check-decl',
        'check-operator',
        'check-separator',
        'check-type',
        'check-typecast' ] } }
[Trace - 10:04:45] Linting: with tslint < version 4
[Trace - 10:04:45] Linting: ended linting
[Trace - 10:04:45] Received notification 'tslint/status'.
Params: {
    "state": 1
}


[Trace - 10:04:45] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c%3A/Projects/Object_List/src/app/pages/login_page/login.ts",
    "diagnostics": []
}


[Trace - 10:04:48] Sending request 'textDocument/codeAction - (5)'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/Projects/Object_List/src/app/pages/login_page/login.ts"
    },
    "range": {
        "start": {
            "line": 24,
            "character": 4
        },
        "end": {
            "line": 31,
            "character": 0
        }
    },
    "context": {
        "diagnostics": []
    }
}
```

**Comment by egamma** [comment]
@edarev thanks for the trace. You are using a tslint version that is older than tslint 4. 

>[Trace - 10:04:45] Linting: with tslint < version 4
[Trace - 10:04:45] Linting: ended linting

Can you reproduce the problem when you install the latest tslint?

**Comment by lijunle** [comment]
I am using VSCode (without insider) and TSLint 5.7, I got the same error.

I have file `temp/tslint.json` relative to my working folder with user settings:

```
"tslint.configFile": "temp/tslint.json",
```

But, it solve the path as:

```
[Info  - 4:47:47 PM] vscode-tslint: Cannot read tslint configuration - 'Could not find config file at: C:\Program Files\Microsoft VS Code\temp\tslint.json'
```

Days ago, it is working fine.

**Comment by egamma** [comment]
@lijunle this is issue #259 and tracked here. I broke the support for relative path names to the config file. The work around is to use an absolute path name.

**Comment by egamma** [comment]
@edarev there was bug when using tslint version < 4.0. This is now fixed and I've published version 1.0.6 with the fix.

**Comment by egamma** [comment]
@willpracht @DjulLau can you please check whether 1.0.6 addresses your issue as well. 1.0.6 fixes issue #257.

**Comment by willpracht** [comment]
@egamma Yeah, let me see if I can get that setup today.

Also, does this require me to use TypeScript 2.x? Or should I be good with TypeScript 1.8?
I'm also running tslint 3.x, will that be an issue or do I need to bump to 5.x?

**Comment by willpracht** [comment]
@egamma Hey, good news!

With the following, things seem to work again!:

tslint: /Users/me/.nvm/versions/node/v4.8.4/bin/tslint : 5.7.0
tsc: /Users/me/.nvm/versions/node/v4.8.4/bin/tsc : 2.5.2
vscode-tslint: 1.0.6
vscode: 1.16.0 (not insiders build)

Locally (via gulp) I'm running TypeScript 1.8.10 and tslint 3.something, so ideally I would be doing the same globally, but I'm afraid to touch it now ;p.

**Comment by DjulLau** [comment]
@egamma Sorry I'm not able to get back in the configuration of my initial issue so I cannot test if that would have fixed it. 
Thanks for your support and for this superb plugin!

**Comment by edarev** [comment]
@egamma Thanks, it works!

**Comment by egamma** [comment]
Thanks for verifying, closing.
